src/Common/Core/Impl/Threading/AsyncReaderWriterLock.Queue.cs

```diff
@@ -134,7 +134,7 @@ private class Queue {
             public LockSource[] Remove(LockSource item) {
                 lock (this) {
                     var next = item.Next;
-                    Interlocked.CompareExchange(ref _head, next, item);
+                    var oldHead = Interlocked.CompareExchange(ref _head, next, item);
```

Comment:
`Interlocked.CompareExchange` is translated into `lock cmpxchg` on multi-core machine, so there is no need to use `volatile` and it is just one operation instead of 
```cs
var oldHead = _head;
if (item == _head) {
    _head = next;
}
```